detox/ios/Detox/DetoxInstrumentsManager.h

```diff
@@ -8,12 +8,13 @@
 
 #import <Foundation/Foundation.h>
 
+
 @interface DetoxInstrumentsManager : NSObject
 
 + (NSURL*)defaultURLForTestName:(NSString*)testName;
 
-- (void)startRecordingAtURL:(NSURL*)URL;
-- (void)continueRecordingAtURL:(NSURL*)URL;
+- (void)startRecordingAtURL:(NSURL*)URL andSamplingInterval:(NSTimeInterval)samplingInterval;
```

Comment:
Instead of passing a sampling interval, the method should receive a `DTXProfilingConfiguration` object with the configuration overridden as needed. Selector name should be `startRecordingWithConfiguration:`; the URL can now be set directly on the DTXProfilingConfiguration object.

Another option is to change this method to `startRecordingAtURL:configuration:` and pass an NSDictionary with config values, and then create the `DTXProfilingConfiguration` internally.

This way we can extend this to support full configuration setting from json config.

Reply:
I though about this as we'll may have many different, but decided to do this when we'll have 3+ params for recording.

Reply:
Let's do it now. It will be needed once we really start using the feature.